src/Pulp/Build.purs

```diff
@@ -90,8 +92,16 @@ go buildType = Action \args -> do
 
   out.log "Build successful."
 
-  shouldBundle <- (||) <$> getFlag "optimise" opts <*> hasOption "to" opts
-  when shouldBundle (bundle args)
+  shouldBundle' <- shouldBundle args
+  when shouldBundle' do
+    failIfUsingEsModulesPsVersion out $ Just
+      "Code path reason: you used the --optimize and/or --to flag(s)"
```

Comment:
Same comment here. Unless this is meant to be internal, I'd like to provide a solution to the user — this sounds like it's just telling them what commands they used. Please let me know if I'm missing something!

```suggestion
      "The --optimize and --to flags can only be used with PureScript projects using CommonJS modules, but your project is using ES modules."
```

Reply:
I don't provide a solution because I'm not sure what the solution should be. My goal in this PR was to unblock the ecosystem migration efforts by getting `pulp` to work on ES modules. I haven't through through what the user story should be for supporting ES modules. For example, do we follow what `spago` will do and start bundling things via `esbuild`? Or do we leave it up to the user to do that? Or do we recommend something like `pulp --then "esbuild bundle --minify" build`? So, I assume we would work further on `pulp` before making a new `v16.0.0` release. However, I thought we could at least make a pre-release for it.

Reply:
My apologies, "solution" wasn't the right word here. I just mean more information they can use to work from beyond just telling them the commands that failed. However, your other comment about the generic error message clears this up.